src/servers/src/http/authorize.rs

```diff
@@ -157,18 +194,8 @@ fn get_influxdb_credentials<B: Send + Sync + 'static>(
         // try v1
         let Some(query_str) = request.uri().query() else { return Ok(None) };
 
-        // TODO(shuiyisong): remove this for performance optimization
-        // `authorize` would deserialize query from urlencoded again
-        let query = match serde_urlencoded::from_str::<HashMap<String, String>>(query_str) {
-            Ok(query_map) => query_map,
-            Err(e) => IllegalParamSnafu {
-                msg: format!("fail to parse http query: {e}"),
-            }
-            .fail()?,
-        };
-
-        let username = query.get("u");
-        let password = query.get("p");
+        let username = extract_helper(query_str, &regex_extractor.influxdb_user_extractor);
+        let password = extract_helper(query_str, &regex_extractor.influxdb_pass_extractor);
```

Comment:
if we want extreme performance we can combine these user/pass regex into one and match only once